.github/workflows/release-please.yaml

```diff
@@ -0,0 +1,287 @@
+      prs: ${{ steps.release.outputs.prs }}
+      paths_released: ${{ steps.release.outputs.paths_released }}
+    steps:
+
+      - name: Run release-please
+        id: release
+        uses: googleapis/release-please-action@v4
+        with:
+          token: ${{ secrets.gh_token }}
+          config-file: ${{ inputs.config }}
+          manifest-file: ${{ inputs.manifest }}
+
+      - name: Slack notification
+        if: failure()
+        uses: matter-labs/zksync-ci-common/.github/actions/slack-notify-release@aba-release-please-support
+        with:
+          webhook: ${{ secrets.slack_webhook }}
+          context: 'Unable execute release-please'
+
+
+  update-cargo-lock:
+    name: Update Cargo.lock
+    runs-on: ubuntu-latest
+    needs: release-please
+    if: ${{ needs.release-please.outputs.prs_created == 'true' && inputs.update-cargo-lock  == 'true' }}
+    strategy:
+      matrix:
+        pr_branch: ${{ fromJson(needs.release-please.outputs.prs).*.headBranchName }}
+      fail-fast: false
+      max-parallel: 1
+    steps:
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ matrix.pr_branch }}
+          token: ${{ secrets.gh_token }}
+          submodules: "recursive"
+
+      # It will automatically check for rust-toolchain file in the repository
+      # and take care of the proper caching to speed up CI.
+      - name: Install Rust toolchain
+        uses: moonrepo/setup-rust@v1
+
+      - name: Update Cargo.lock
+        shell: 'bash -ex {0}'
+        run: |
+          BRANCH="${{ matrix.pr_branch }}"
+          if [[ ${BRANCH} == *components* ]]; then
+            COMPONENT="${BRANCH##*--}"
+            COMPONENT_PATH=$(jq -r --arg component ${COMPONENT} '.packages | to_entries[] | select(.value.component == $component) | .key' ${{ inputs.config }})
+            ( cd "${COMPONENT_PATH}" && cargo update --workspace )
+          else
+            for WORKSPACE in ${{ inputs.workspace-dirs }} ; do
+              ( cd "${WORKSPACE}" && cargo update --workspace )
+            done
+          fi
+          if ! git diff --exit-code --quiet; then
+            git config user.name "${{ inputs.git-user-name }}"
+            git config user.email "${{ inputs.git-user-email }}"
+            git add .
+            git commit -m "${{ inputs.cargo-lock-commit-message }}"
+            git push --set-upstream origin ${{ matrix.pr_branch }}
+          fi
+
+      - name: Slack notification
+        if: failure()
+        uses: matter-labs/zksync-ci-common/.github/actions/slack-notify-release@aba-release-please-support
+        with:
+          webhook: ${{ secrets.slack_webhook }}
+          context: 'Unable to update Cargo.lock'
+
+
+  publish-crates:
+    needs: release-please
+    name: Publish to crates.io
+    runs-on: ubuntu-latest
+    if: ${{ needs.release-please.outputs.releases_created == 'true' && inputs.publish-to-crates-io == 'true' }}
```

Comment:
What kind of worries me here is idempotency. What do we do if publishing fails due to a transient error (e.g. rate limiting)?
It should be possible to just re-run the workflow.

Is it possible now?

Reply:
As this is a separate job, it would be good enough to just restart the failed job instead of the whole workflow, and it should publish them if no issues.

But I believe it might be a good idea to move publishing into its own action, and create an additional workflow that we can manually start for publishing a component for urgent cases. I think it can be useful when `release-please` doesn't trigger publishing job, but we actually need it due some previous errors.